src/base/TypeChecker.ml

```diff
@@ -329,15 +329,25 @@ module ScillaTypechecker (SR : Rep) (ER : Rep) = struct
                (mk_qual_tp t, rep) ),
              remaining_gas )
     | TFun (tvar, body) ->
-        let tenv' = TEnv.addV (TEnv.copy tenv) tvar in
-        let%bind ((_, (bt, _)) as typed_b), remaining_gas =
-          type_expr tenv' body remaining_gas
-        in
-        let typed_tvar = add_type_to_ident tvar bt in
-        pure
-        @@ ( ( TypedSyntax.TFun (typed_tvar, typed_b),
-               (mk_qual_tp (PolyFun (get_id tvar, bt.tp)), rep) ),
-             remaining_gas )
+        let id = get_id tvar in
+        (* XXX this is a workaround for alpha-renaming *)
```

Comment:
Are we happy with this as a permanent fix? If not, we should create an issue for the actual fix (which is probably on-the-fly renaming).

Reply:
Here is the issue: https://github.com/Zilliqa/scilla/issues/794